app/static/src/app/store/code/mutations.ts

```diff
@@ -0,0 +1,12 @@
+import { MutationTree } from "vuex";
+import { CodeState } from "./state";
+
+export enum CodeMutation {
+    SetCode = "SetCode"
+}
+
+export const mutations: MutationTree<CodeState> = {
+    [CodeMutation.SetCode](state: CodeState, payload: string[]) {
+        state.code = [...payload];
```

Comment:
What are you trying to achieve spreading the payload?

Reply:
I used the spread since the payload is the default code which i don't want to get modified directly from the code editor - but actually, having gone on and implemented this, I think we actually will be calling the mutation to set code each time with a different array, so shouldn't need this anymore. I'll remove.